language/move-lang/src/expansion/dependency_ordering.rs

```diff
@@ -46,44 +62,79 @@ enum DepType {
     Friend,
 }
 
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd)]
+enum ModuleOrScriptIdent {
+    Module(ModuleIdent),
+    Script(String),
+}
+
 struct Context<'a> {
     modules: &'a UniqueMap<ModuleIdent, E::ModuleDefinition>,
-    // A union of uses and friends:
+    // A union of uses and friends for modules (used for cyclyc dependency checking)
     // - if A uses B,    add edge A -> B
     // - if A friends B, add edge B -> A
-    neighbors: BTreeMap<ModuleIdent, BTreeMap<ModuleIdent, BTreeMap<DepType, Loc>>>,
-    current_module: Option<ModuleIdent>,
+    // NOTE: neighbors of scripts are not tracked by this field, as nothing can depend on a script
+    // and a script cannot declare friends. Hence, is no way to form a cyclic dependency via scripts
+    module_neighbors: BTreeMap<ModuleIdent, BTreeMap<ModuleIdent, BTreeMap<DepType, Loc>>>,
+    // A summary of neighbors keyed by module or script
+    neighbors_by_unit: BTreeMap<ModuleOrScriptIdent, BTreeSet<E::Neighbor>>,
+    // The module or script we are currently exploring
+    current: Option<ModuleOrScriptIdent>,
```

Comment:
if we rename this to NodeIdent, it can become `current_node` which reads a bit better maybe

Reply:
Renamed, and also renamed `neighbors_by_unit` into `neighbors_by_node`.